app/models/manageiq/providers/redhat/infra_manager/api_integration.rb

```diff
@@ -19,32 +20,38 @@ def connect(options = {})
     unless options[:skip_supported_api_validation] || supports_the_api_version?(version)
       raise "version #{version} of the api is not supported by the provider"
     end
-    # If there is API path stored in the endpoints table and use it:
+
+    # If the API path is stored in the endpoints table then use it:
     path = options[:path] || default_endpoint.path
     _log.info("Using stored API path '#{path}'.") unless path.blank?
 
-    server   = options[:ip] || address
-    port     = options[:port] || self.port
-    username = options[:user] || authentication_userid(options[:auth_type])
-    password = options[:pass] || authentication_password(options[:auth_type])
-    service  = options[:service] || "Service"
+    # Prepare the options to call the method that creates the actual connection:
+    connect_options = {
+      :server     => options[:ip] || address,
+      :port       => options[:port] || self.port,
+      :username   => options[:user] || authentication_userid(options[:auth_type]),
+      :password   => options[:pass] || authentication_password(options[:auth_type]),
+      :service    => options[:service] || "Service",
+      :verify_ssl => default_endpoint.verify_ssl,
+      :ca_certs   => default_endpoint.certificate_authority
+    }
 
     # Starting with version 4 of oVirt authentication doesn't work when using directly the IP address, it requires
     # the fully qualified host name, so if we received an IP address we try to convert it into the corresponding
     # host name:
     if resolve_ip_addresses?
-      resolved = resolve_ip_address(server)
-      if resolved != server
-        _log.info("IP address '#{server}' has been resolved to host name '#{resolved}'.")
+      resolved = resolve_ip_address(connect_options[:server])
+      if resolved != connect_options[:server]
+        _log.info("IP address '#{connect_options[:server]}' has been resolved to host name '#{resolved}'.")
         default_endpoint.hostname = resolved
-        server = resolved
+        connect_options[:server] = resolved
```

Comment:
We're enabling TLS validation mainly to protect against MITM.
Are we enforcing DNSSEC?  Assuming not, DNS is all unsigned plaintext, trivially subvertable by a man in the middle.
Normally TLS makes DNS compromise not a threat (well just a DoS) because the wrong IP will not be able to present the right cert, but in this specific scenario, DNS becomes the weak link.

User should not expect "do what I mean with this IP address" to provide security.  I propose making that clear by not allowing it.  User will still be able to specify domain name OR turn off validation.
(It's possibly OK to allow IP address with a custom CA, not sure.)